### PR TITLE
Nitsmell : Adding new code smells and print console updated

### PR DESCRIPTION
**Pull request by Delja**
Add :
- New "no abstract implementation" code smell
- Toolcontext option to print one by one a detected codesmell
Update :
- Print console improve
- Feature envy codesmell prepose a move help

### CODE REVIEW

**Comment by privat** [comment]
ok to test

**Comment by privat** [comment]
missing signed-off-by

File: src/metrics/codesmells_metrics.nit [hunk 1]
```diff
@@ -159,15 +184,20 @@ class LargeClass
...
 	redef fun collect(mclassdef, model_builder): Bool do
-		number_attribut = mclassdef.collect_intro_and_redef_mattributes(model_builder.model.private_view).length
-		# get the number of methods (subtract the get and set of attibutes with (numberAtribut*2))
-		number_method = mclassdef.collect_intro_and_redef_methods(model_builder.model.private_view).length
-		return number_method.to_f > phase.average_number_of_method and number_attribut.to_f > phase.average_number_of_attribute
+		self.number_attribut = mclassdef.collect_intro_and_redef_mattributes(model_builder.model.private_view).length
+		# Get the number of methods (Accessor include) (subtract the get and set of attibutes with (numberAtribut*2))
+		self.number_method = mclassdef.collect_intro_and_redef_methods(model_builder.model.private_view).length
+		self.score_calcul
+		return self.number_method.to_f > phase.average_number_of_method and self.number_attribut.to_f > phase.average_number_of_attribute
```
Comment (jcbrinfo) [inline src/metrics/codesmells_metrics.nit]: Python? 😄

**Comment by privat** [comment]
things to fix

* [ ] whitespaces http://gresil.org/jenkins/job/CI_github/5342/testReport/junit/cmd/check/whitespaces/
* [ ] signed off by http://gresil.org/jenkins/job/CI_github/5342/testReport/junit/cmd/check/signed_off_by/

**Comment by privat** [comment]
Please, fix the tests.

File: src/metrics/codesmells_metrics.nit [hunk 2]
```diff
@@ -109,32 +112,49 @@ class BadConceptionFinder
...
+	# Collect code smell with selected toolcontext option
 	fun collect do
 		var bad_conception_elements = new Array[BadConception]
-		bad_conception_elements.add(new LargeClass(phase))
-		bad_conception_elements.add(new LongParameterList(phase))
-		bad_conception_elements.add(new FeatureEnvy(phase))
-		bad_conception_elements.add(new LongMethod(phase))
+		# Check toolcontext option
+		if phase.toolcontext.opt_feature_envy.value or phase.toolcontext.opt_all.value then bad_conception_elements.add(new FeatureEnvy(phase))
+		if phase.toolcontext.opt_long_method.value or phase.toolcontext.opt_all.value then 	bad_conception_elements.add(new LongMethod(phase))
```
Comment (privat) [inline src/metrics/codesmells_metrics.nit]: why the weird spacing?

Comment (Delja) [inline src/metrics/codesmells_metrics.nit]: It was a keying errors

**Pull request merged**